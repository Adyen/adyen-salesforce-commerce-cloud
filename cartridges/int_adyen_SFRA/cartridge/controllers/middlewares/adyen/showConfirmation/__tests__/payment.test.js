"use strict";

var OrderMgr = require('dw/order/OrderMgr');
var _require = require('../payment'),
  handlePaymentInstruments = _require.handlePaymentInstruments;
var req;
beforeEach(function () {
  req = {
    querystring: {
      redirectResult: 'mocked_redirect_result'
    }
  };
});
describe('Payment', function () {
  it('should handle payment instruments with redirect result', function () {
    var paymentInstruments = OrderMgr.getPaymentInstruments();
    var result = handlePaymentInstruments(paymentInstruments, {
      req: req
    });
    expect(result).toMatchSnapshot();
  });
  it('should handle payment instruments with payload', function () {
    var paymentInstruments = OrderMgr.getPaymentInstruments();
    req.querystring = {
      payload: 'mocked_payload'
    };
    var result = handlePaymentInstruments(paymentInstruments, {
      req: req
    });
    expect(result).toMatchSnapshot();
  });
});