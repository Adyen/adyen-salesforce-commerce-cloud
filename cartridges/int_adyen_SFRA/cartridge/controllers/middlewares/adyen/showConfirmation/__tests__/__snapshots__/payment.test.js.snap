// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment should handle payment instruments with payload 1`] = `
Object {
  "adyenPaymentInstrument": Object {
    "custom": Object {
      "adyenAction": "mocked_adyen_action",
      "adyenMD": "mocked_adyen_MD",
      "adyenPaymentData": "mocked_adyen_payment_data",
      "adyenRedirectURL": "https://some_mocked_url/signature",
    },
  },
  "details": Object {
    "payload": "mocked_payload",
  },
  "paymentData": "mocked_adyen_payment_data",
}
`;

exports[`Payment should handle payment instruments with redirect result 1`] = `
Object {
  "adyenPaymentInstrument": Object {
    "custom": Object {
      "adyenAction": "mocked_adyen_action",
      "adyenMD": "mocked_adyen_MD",
      "adyenPaymentData": "mocked_adyen_payment_data",
      "adyenRedirectURL": "https://some_mocked_url/signature",
    },
  },
  "details": Object {
    "redirectResult": "mocked_redirect_result",
  },
  "paymentData": "mocked_adyen_payment_data",
}
`;
